### ISSUE TITLE
cilium operator unable to process empty tags or tags with an @ symbol or tags with spaces

### ISSUE THREAD

**Issue by allenmun197** [cilium/cilium#19793]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

we're seeing an issue with the cilium-operator, when we spin up a kops cluster, the `--eni-tags` flag is used and when passing an @ symbol or a empty key tag like keepname=, cilium errors out with
```
level=info msg="Started gops server" address="127.0.0.1:9891" subsys=cilium-operator
level=fatal msg="unable to parse eni-tags: 'KubernetesCluster=test-cluster,OwnerContact=randomtestemail@gmail.com,keepname=' is not formatted as key=value,key1=value1" subsys=option
```
These seem to be valid tags when adding them as aws tags but it seems like the cilium operator is unable to handle it.

### Cilium Version

# cilium version
Client: 1.10.10 8a77a4a 2022-04-15T08:30:54-07:00 go version go1.16.15 linux/amd64
Daemon: 1.10.10 8a77a4a 2022-04-15T08:30:54-07:00 go version go1.16.15 linux/amd64

### Kernel Version

# uname -a
Linux ip-10-206-248-11.... 5.10.109-104.500.amzn2.x86_64 #1 SMP Wed Apr 13 20:31:43 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.9", GitCommit:"6df4433e288edc9c40c2e344eb336f63fad45cd2", GitTreeState:"clean", BuildDate:"2022-04-13T19:52:02Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ungureanuvladvictor** [comment]
@allenmunC1 -- if you try to deploy just with the tag that has `@` will it work ?

**Comment by sayboras** [comment]
thanks for reporting this, seems like there either empty value and @ character is not allowed right now.

As a work around, can you try to pass value with json format e.g. { "k1": "v1", "k2": "" } ? Thanks.

**Comment by allenmunC1** [comment]
> thanks for reporting this, seems like there either empty value and @ character is not allowed right now.
> 
> As a work around, can you try to pass value with json format e.g. { "k1": "v1", "k2": "" } ? Thanks.

Unfortunately, we are using kops that automatically formats and deploys `cilium-operator` with the `--eni-tags` as a string.  So we cannot change this easily

**Comment by sayboras** [comment]
you can still pass json string e.g. --eni-tags = '{ \k1\": \"v1\", \"k2\": \"\" }' with a proper escape.

**Comment by allenmunC1** [comment]
Is it possible to re-open this issue, it seems that this was not fixed in v1.10.12 which https://github.com/cilium/cilium/pull/19794 was back-ported to.

Still getting
```
level=info msg="Started gops server" address="127.0.0.1:9891" subsys=cilium-operator
level=fatal msg="unable to parse eni-tags: 'BackupPlan=D2P-EBS-Default,KubernetesCluster=my-test-cluster.test-domain.com,OwnerContact=test@test.com,keepname=true' is not formatted as key=value,key1=value1" subsys=option
```

when using the image `quay.io/cilium/operator:v1.10.12` in our cilium operator pods

**Comment by sayboras** [comment]
sorry to hear that, can you try with json format for the time being ?
